corehq/apps/users/models.py

```diff
@@ -1126,6 +1128,10 @@ def get(stale, raise_if_none):
     def clear_quickcache_for_user(self):
         from corehq.apps.hqwebapp.templatetags.hq_shared_tags import _get_domain_list
         self.get_by_username.clear(self.__class__, self.username)
+        self.get_by_user_id.clear(self.__class__, self.user_id)
+        all_domains = self._original_domains | set(self.get_domains())
+        for domain in all_domains:
+            self.get_by_user_id.clear(self.__class__, self.user_id, domain)
```

Comment:
i think we're also going to have to set _original_domains = self.get_domains() on every save in case multiple operations are happening on the user in a row